### PR TITLE
#41 implemented find method

### PR DESCRIPTION
**Pull request by ammaratef45**
PR for #41 
  - Implemented the pay method.
  - Changed the pay in wallet to return result instead of void.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#43](https://github.com/amihaiemil/zold-java-client/issues/43) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#43](https://github.com/amihaiemil/zold-java-client/issues/43) is assigned to @denis-konovalyenko[/z](https://www.0crat.com/u/denis-konovalyenko), here is [why](https://www.0crat.com/footprint/GFCHY7NQG/52b1a011-c697-4428-b918-ea26aa52a02e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by denis-konovalyenko** [comment]
@ammaratef45 , could you please take a look at the code style errors the qulice produces (I guess that was due to your lines reformatting - and that is another thing to bear in mind - not to touch the code which was written before and it works as needed):
```
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:67: @param tag description should start with capital letter [RegexpSingleline]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:77: Line is longer than 80 characters (found 90). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:79: Line is longer than 80 characters (found 119). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:87: Line is longer than 80 characters (found 100). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:90: Line is longer than 80 characters (found 96). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:97: Line is longer than 80 characters (found 102). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:100: Line is longer than 80 characters (found 95). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:101: Line is longer than 80 characters (found 112). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:103: Line is longer than 80 characters (found 94). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:106: Line is longer than 80 characters (found 104). [LineLength]
[ERROR] /home/travis/build/amihaiemil/zold-java-client/src/main/java/com/amihaiemil/zold/RtWallet.java:113: Line is longer than 80 characters (found 105). [LineLength]
```
This excerpt is from the Travice CI build log.

File: src/main/java/com/amihaiemil/zold/RtWallet.java [hunk 1]
```diff
@@ -70,83 +74,61 @@
 
     @Override
     public String getId() throws IOException {
-        final HttpGet walletId = new HttpGet(
```
Comment (denis-konovalyenko) [inline src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45 , could you please revert back the irrelevant to the `find` method changes throughout the whole class?

Comment (ammaratef45) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @denis-konovalyenko :disappointed: I will get rid of VS Code :laughing: 
Sorry, I will revert all of these changes and resubmit the code.

**Comment by ammaratef45** [comment]
@denis-konovalyenko Please check now

File: src/main/java/com/amihaiemil/zold/RtWallet.java [hunk 2]
```diff
@@ -143,10 +146,27 @@ public void pay(
...
 
     @Override
-    public void find(final String id, final String details) {
-        throw new UnsupportedOperationException(
-            "Not yet implemented. If you can contribute please, do it at "
-            + "https://github.com/amihaiemil/zold-java-client"
+    public JsonArray find(
+            final String id, final String details
+    ) throws IOException, URISyntaxException {
+        URIBuilder builder = new URIBuilder(this.baseUri.toString() + "/find");
```
Comment (denis-konovalyenko) [inline src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45, this can be final I guess. But that's nitpicking...

**Comment by denis-konovalyenko** [comment]
@ammaratef45 , @amihaiemil , the PR looks good and can be merged.

**Comment by denis-konovalyenko** [comment]
@rultor merge

**Comment by denis-konovalyenko** [comment]
@0crat , wait for @amihaiemil decision on this.

**Comment by 0crat** [comment]
> @0crat , wait for @amihaiemil decision on this. ([here](https://github.com/amihaiemil/zold-java-client/issues/43#issuecomment-467961422))

@denis-konovalyenko Can't understand ",", try one of these:
  - `assign` Assign a performer to this issue
  - `boost` Set a boost for the job
  - `continue` Remove a job's impediment
  - `hello` Just say hello
  - `in` Register this issue in scope (WBS)
  - `out` Close the order and remove this job from scope
  - `pay` Pay a user some extra cash
  - `quality` Review a task
  - `resign` Resign from this issue
  - `status` Check the status of the job
  - `version` Print current version of the bot
  - `wait` Register an impediment for a job

**Comment by denis-konovalyenko** [comment]
@0crat  wait for @amihaiemil decision on this.

**Comment by 0crat** [comment]
> @0crat wait for @amihaiemil decision on this. ([here](https://github.com/amihaiemil/zold-java-client/issues/43#issuecomment-467963916))

@denis-konovalyenko It's a code review job [#43](https://github.com/amihaiemil/zold-java-client/issues/43), can't put it on hold

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
The job [#43](https://github.com/amihaiemil/zold-java-client/issues/43) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @denis-konovalyenko[/z](https://www.0crat.com/u/denis-konovalyenko)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**